src/app/shell/details/achievements/achievements.component.ts

```diff
@@ -36,7 +36,7 @@ export class AchievementsComponent implements OnInit {
   }  
 
   private getAchievements(): void {
-    // this.store.dispatch(new GetAchievementsByWorkshopId(this.workshop.id));
+    this.store.dispatch(new GetAchievementsByWorkshopId(this.workshop?.id));
```

Comment:
why there is  "?" 
Don't think it will work, if there going to be undefined, so the request will fail. you should do such a request only when the workshop already exists. for example in subscription